### PR TITLE
Adding more functions all, any, get_dtype, and pass through functions.

### PR DESCRIPTION
**Pull request by ankur-gupta**

### CODE REVIEW

File: flicker/flicker.py [hunk 1]
```diff
@@ -204,6 +204,21 @@ def _reset(self, df):
...
+    def _validate_column_name(self, name, names=None, dataframe_name=None):
+        if not isinstance(name, six.string_types):
+            msg = ('column name must be of type str but you provided '
+                   'type = {}').format(str(type(name)))
+            raise TypeError(msg)
+        if names is None:
+            names = self._df.columns
+        if name not in names:
+            if dataframe_name is None:
+                msg = 'column name "{}" not found'
```
Comment (yulyt) [inline flicker/flicker.py:216]: I think you're missing a `msg.format(name)` for this?

File: flicker/flicker.py [hunk 2]
```diff
@@ -204,6 +204,21 @@ def _reset(self, df):
         self._nrows = None
         self._ncols = None
 
+    def _validate_column_name(self, name, names=None, dataframe_name=None):
```
Comment (yulyt) [inline flicker/flicker.py:207]: Add docstring?

File: flicker/flicker.py [hunk 3]
```diff
@@ -204,6 +204,21 @@ def _reset(self, df):
...
+        if not isinstance(name, six.string_types):
+            msg = ('column name must be of type str but you provided '
+                   'type = {}').format(str(type(name)))
+            raise TypeError(msg)
+        if names is None:
+            names = self._df.columns
+        if name not in names:
```
Comment (yulyt) [inline flicker/flicker.py:214]: If `names` is None, this will raise an exception. Was an `elif` meant to be used here?

File: flicker/flicker.py [hunk 4]
```diff
@@ -985,7 +1000,11 @@ def value_counts(self, names, normalize=False, sort=True, ascending=False,
         if isinstance(names, six.string_types):
             names = [names]
         for name in names:
```
Comment (yulyt) [inline flicker/flicker.py:1002]: Could you use the `_validate_column_name` method here too?

File: flicker/flicker.py [hunk 5]
```diff
@@ -1409,16 +1463,16 @@ def join(self, other, on=None, how='inner',
...
-            if not isinstance(name, six.string_types):
-                msg = ('column names must be of type str but you provided '
-                       'type = {}')
-                msg = msg.format(str(type(name)))
-                raise TypeError(msg)
-            if name not in names:
-                msg = ('column name "{}" not found in the {} dataframe')
-                msg = msg.format(name, dataframe)
-                raise KeyError(msg)
+        # def _validate_column_name(name, names, dataframe):
```
Comment (yulyt) [inline flicker/flicker.py:1466]: rm?

File: flicker/flicker.py (hunk 1 above)
Comment (ankur-gupta) [reply flicker/flicker.py:216]: You're right. Thanks!

File: flicker/flicker.py (hunk 2 above)
Comment (ankur-gupta) [reply flicker/flicker.py:207]: Added!

File: flicker/flicker.py (hunk 4 above)
Comment (ankur-gupta) [reply flicker/flicker.py:1022]: Thanks!

File: flicker/flicker.py (hunk 5 above)
Comment (ankur-gupta) [reply flicker/flicker.py]: Fixed, thanks :)

**Comment by ankur-gupta** [comment]
@yulyt Thanks for you review! I have addressed all of your comments. I've also added tests to increase the coverage from ~52% to ~61%. More tests need to be written (especially for `value_counts()` which has a known issue) but we're making progress. 

If you can, please take another look. If not, it's okay. I'll merge it later today.

Once we merge this PR, I would like to add a `CONTRIBUTING.md` file and replace my name with `Flicker Contributors` in all copyright notices. We already have an Apache 2.0 license (same as spark).

**Pull request merged**